src/Transactions/Builder/EvmCallBuilder.php

```diff
@@ -4,31 +4,49 @@
 
 namespace ArkEcosystem\Crypto\Transactions\Builder;
 
-use ArkEcosystem\Crypto\Transactions\Types\EvmCall;
+use ArkEcosystem\Crypto\Configuration\Fee;
+use ArkEcosystem\Crypto\Configuration\Network;
+use ArkEcosystem\Crypto\Enums\TypeGroup;
+use ArkEcosystem\Crypto\Enums\Types;
+use ArkEcosystem\Crypto\Identities\PrivateKey;
+use ArkEcosystem\Crypto\Transactions\Transaction;
 
-class EvmCallBuilder extends AbstractTransactionBuilder
+class EvmCallBuilder
 {
-    /**
-     * Create a new EVM call transaction instance.
-     */
+    public Transaction $transaction;
+
     public function __construct()
     {
-        parent::__construct();
+        $this->transaction                    = new Transaction();
 
-        $this->transaction->data['asset'] = [
-            'evmCall' => [
-                'gasLimit' => 1000000,  // Default gas limit
-                'payload'  => '',       // EVM code in hex format
+        $this->transaction->data = [
+            'type'            => Types::EVM_CALL->value,
+            'typeGroup'       => TypeGroup::CORE,
+            'amount'          => '0',
+            'senderPublicKey' => '',
+            'fee'             => Fee::get(Types::EVM_CALL->value),
+            'nonce'           => '1',
+            'version'         => 1,
+            'network'         => Network::get()->pubKeyHash(),
+            'asset'           => [
+                'evmCall' => [
+                    'gasLimit' => 1000000,  // Default gas limit
+                    'payload'  => '',       // EVM code in hex format
```

Comment:
let's discuss this part, as the payload is what decides the underlying logic that the transaction will apply.

The way this currently works is that the user has to create the payload, then pass the payload to the builder to get it executed on chain. Although that works, it means that the user first has to figure out the required payload and construct it manually, while that should be the job of the SDK (to make it easier for the user to work with the network).

Therefore I'd suggest that we still add transaction builders, but underneath they all make use of the EVM call transaction type. The builders will however make it easier to pass the usual parameters for each of the supported transaction "types". For example:
- transfer: here you'd call `->amount()` and `->recipient()` on to set these in addition to the payload (which is empty for a transfer)
- vote: this would have a parameter to define the vote/unvote/swap vote. All the payload logic would be handled internally as those things are known to use (recipient contract, abi, function name, args)
- other types (later, but similar idea)

The builders would also keep their usual methods ( `->nonce()`, etc) but having these abstractions will make it easier for the user to work with the evm calls without having to construct raw payloads. Of course they can still provide raw payloads by using the EVM Call builder directly and passing it through the `->payload()` method whenever they need to make a call to a contract that's not one of ours.

I think this will make the SDK easier to work with as all the contracts that we deploy and know about can be included without putting the burden on the user to figure out the right method and contract address to use.